setuptools/command/easy_install.py

```diff
@@ -2079,7 +2079,7 @@ class ScriptWriter:
 
             if __name__ == '__main__':
                 sys.argv[0] = re.sub(r'(-script\.pyw?|\.exe)?$', '', sys.argv[0])
-                for entry_point in distribution(%(spec)r).entry_points:
+                for entry_point in distribution(%(spec)r.split('==')[0]).entry_points:
```

Comment:
I have no context on setuptools test situation etc., but can a test be added to ensure this does not happen again?

Reply:
I'm going to defer the tests for now, as I'm not confident I can build one reliably and quickly. I do agree it would be good to have, although the long-term solution will be for editable installs to have their scripts generated with distlib, same as pip, without the package resources requirement.